pkg/network/eventloop.go

```diff
@@ -113,110 +80,40 @@ func (el *eventLoop) registerRead(conn *connection, handler *connEventHandler) e
 	}
 
 	// register
-	el.poller.Start(read, el.readWrapper(read, handler))
-
-	el.mu.Lock()
-	//store
-	el.conn[conn.id] = &connEvent{
-		read: read,
-	}
-	el.mu.Unlock()
-	return nil
-}
-
-func (el *eventLoop) registerWrite(conn *connection, handler *connEventHandler) error {
-	// handle write
-	write, err := netpoll.HandleFile(conn.file, netpoll.EventWrite|netpoll.EventOneShot)
+	err = el.poller.Start(read, el.readWrapper(conn, read, handler))
 	if err != nil {
+		log.DefaultLogger.Errorf("[network] registerRead failed err : %v, addr: %v", err, conn.RemoteAddr())
 		return err
 	}
 
-	// register
-	el.poller.Start(write, el.writeWrapper(write, handler))
-
-	el.mu.Lock()
 	//store
-	el.conn[conn.id] = &connEvent{
-		write: write,
+	conn.ev = &connEvent{
+		read: read,
 	}
-	el.mu.Unlock()
 	return nil
 }
 
-func (el *eventLoop) unregister(id uint64) {
-
-	if event, ok := el.conn[id]; ok {
-		if event.read != nil {
-			el.poller.Stop(event.read)
-		}
-
-		if event.write != nil {
-			el.poller.Stop(event.write)
-		}
-
-		el.mu.Lock()
-		delete(el.conn, id)
-		el.mu.Unlock()
-	}
-
-}
-
-func (el *eventLoop) unregisterRead(id uint64) {
-	if event, ok := el.conn[id]; ok {
-		if event.read != nil {
-			el.poller.Stop(event.read)
-		}
-
-		el.mu.Lock()
-		delete(el.conn, id)
-		el.mu.Unlock()
-	}
-}
-
-func (el *eventLoop) unregisterWrite(id uint64) {
-	if event, ok := el.conn[id]; ok {
-		if event.write != nil {
-			el.poller.Stop(event.write)
+func (el *eventLoop) unregisterRead(conn *connection) {
+	if conn.ev != nil {
+		err := el.poller.Stop(conn.ev.read)
+		if err != nil {
+			log.DefaultLogger.Errorf("[unregisterRead] failed, %v", err)
 		}
-
-		el.mu.Lock()
-		delete(el.conn, id)
-		el.mu.Unlock()
 	}
 }
 
-func (el *eventLoop) readWrapper(desc *netpoll.Desc, handler *connEventHandler) func(netpoll.Event) {
+func (el *eventLoop) readWrapper(conn *connection, desc *netpoll.Desc, handler *connEventHandler) func(netpoll.Event) {
```

Comment:
It‘s affect performance that schedule method would block if no worker goroutine is available.